### ISSUE TITLE
"suitable is not defined" Error when using with Cider

### ISSUE THREAD

**Issue by DogLooksGood** [rksm/clj-suitable#1]
Thanks for the great work!
I have a try by setuping a minimal project with figwheel-main.
Everything works perfect, except there's an error in browser console.
```
REPL eval error ReferenceError: "suitable is not defined"
```
But the auto-completion actually works!

**Comment by rksm** [comment]
Yep, I see that as well. This comes from [js-properties-of-object](https://github.com/rksm/clj-suitable/blob/master/src/main/suitable/js_completions.cljc#L28) that does a cljs eval to require `suitable.js-introspection` which is the namespace with the functions to enumerate the obj props. When we call that for the first time, the require runs but doesn't seem to update the environment. This means that the subsequent call to `suitable.js-introspection/property-names-and-types` fails. Only when it runs the next time it is loaded.

I'm not really familiar with the cljs implementaion and don't know what the proper way to deal with cljs API for loading requirements is, played around with `cljs.repl/load-namespace` and such but this doesn't seem to work / update the repl-env. Any ideas are welcome.

Anyway, looking into a fix...

**Comment by DogLooksGood** [comment]
will preload work for this? It can be done by middleware or configuring cljs build by user.

**Comment by rksm** [comment]
Yeah, preload will work but I would like to keep the setup as simple as possible. Since we already get control in the middleware function we should be able to coordinate loading the remaining deps from there... Looking through the docs, `load-namespace` _should_ work (https://clojurescript.org/reference/repl#code-loading-in-the-browser). Hmm, need to play around with it some more...

**Comment by rksm** [comment]
Ahhhhh... `load-namespace` _and_ `require` :)

**Comment by rksm** [comment]
Aaaaand down the rabbit hole...

So, if you look deep enough under the hood then things turn out not to be so simple... even in Clojure. Lesson learned: Don't believe everything people tell you ;)

So I tried things out with a plain cljs browser repl. The setup looks something like this


```clojure
(require '[cljs.build.api :as b])
(require '[cljs.env :as env])

(def cenv (env/default-compiler-env))

(def opts {:output-to "target/public/out/main.js"
           :output-dir "target/public/out"
           :optimizations :none
           :main "hello-world.core"
           :verbose true
           :compiler-stats true
           :print-input-delimiter true})

(def browser-env (browser/repl-env :port 9300
                                   :launch-browser false
                                   ;; :working-dir
                                   :static-dir ["." "out/" "target/public/"]
                                   :src "src"
                                   :compiler-env cenv))

(cljs.repl/repl* browser-env opts)
```

I make sure that suitable is on the class path and start clj. This directly creates simple browser repl that gets setup once I open a web browser on localhost:9300. I also connect to the same clj process via an nrepl server I sideloaded in to mess with the objects defined above.

When I evaluate this beauty 

```clojure
(binding [cljs.env/*compiler* cenv
          cljs.analyzer/*cljs-ns* "cljs.user"]
  (cljs.repl/load-namespace browser-env 'suitable.js-introspection opts)
  (cljs.repl/evaluate browser-env "<suitable>" 1 "goog.require(\"suitable.js_introspection\")")
  (cljs.repl/evaluate-form browser-env @cenv "<suitable>" '(cljs.core/prn (suitable.js-introspection/property-names-and-types js/console "lo"))))
```

things work as expected. All Good, right?

Hold your horses little FP programmer.

The same thing stops working when using figwheel instead of the plain repl. The last `evaluate-form` throws `repl.cljc:372 REPL eval error ReferenceError: suitable is not defined`. Menno! 

So why? Plain cljs doesn't provide a real inline require b/c as it turns out the sementics a rather hard to implement in pure JS. Figwheel doesn't care and does it anyway. So when you call require in figwheel -- or even plain and pure `goog.require("suitable.js_introspection")` as we did above(!!!) -- eventually `goog.importScript("https?://host:XXX/cljs-out/dev/goog/../suitable/js_introspection.js")` will get called. And this gets [monkey patched by figwheel with queued-file-reload](https://github.com/bhauman/figwheel-repl/blob/master/src/figwheel/repl.cljc#L210). And as we can see in there, our wonderfully simple require just turned into an async mess. So even though the call succeeds, the required ns won't be there immediately. So basically our approach from above just doesn't work.

To be fair, goog.require would normally not work at runtime at all. Cljs already provides a patched variant for it. On the other hand, it would be possible - even in JavaScript - to implement file [loading synchronous using XMLHttpRequest](https://developer.mozilla.org/en-US/docs/Web/API/XMLHttpRequest/Synchronous_and_Asynchronous_Requests#Synchronous_request). IMHO is that what figwheel should do. But yeah.

Anyway, I guess using preloads is the way to go here. ¯\_(ツ)_/¯
